app/models/responses/StatementSearchFailureNotificationErrorResponse.scala

```diff
@@ -58,17 +58,19 @@ object StatementSearchFailureNotificationErrorResponse {
                                    errorDetailMsg: String,
                                    errorMsgList: Seq[String]): Seq[String] = {
     statementRequestID.fold(errorMsgList)(stReqId => Seq(
-      if (errorCode == ErrorCode.code500)
-        if (errorDetailMsg.isEmpty) ErrorMessage.technicalErrorDetail(stReqId)
-        else errorDetailMsg
-      else ErrorMessage.invalidStatementReqIdDetail(stReqId))
+      if (errorCode == ErrorCode.code500) {
```

Comment:
I think we can simplify this by using an if, else if, else statement.
Like if (code500 & is empty) do technicalerrordetails else if isEmpty do errordetailsmsg, else invalidstatemnet

Reply:
Done, logic has been moved to a separate method